protostar/cli/signable_command_mixin.py

```diff
@@ -0,0 +1,170 @@
+            )
+            # pylint: disable=invalid-name
+        r, s = message_signature(msg_hash=tx_hash, priv_key=self.private_key)
+        return [r, s]
+
+
+class SignableCommandMixin:
+    @staticmethod
+    def get_signer(
+        args: Any,
+        network_config: NetworkConfig,
+        logger: Logger,
+    ) -> Optional[
+        BaseSigner
+    ]:  # TODO(arcticae): Make it non-optional in some time in the future
+        if args.signer_class:
+            *module_names, class_name = args.signer_class.split(".")
+            module = ".".join(module_names)
+            signer_module = importlib.import_module(module)
+            signer_class = getattr(signer_module, class_name)
+            SignableCommandMixin._validate_signer_interface(signer_class)
+            return signer_class()
+
+        private_key_str = None
+        if args.private_key_path:
+            with open(args.private_key_path, encoding="utf-8") as file:
+                private_key_str = file.read()
+
+        if args.private_key_path and not args.account_address:
+            raise ProtostarException(
+                "account-address option has to be specified when using private key for signing"
+            )
+
+        if not private_key_str:
+            private_key_str = os.environ.get(PRIVATE_KEY_ENV_VAR_NAME)
+
+        if args.account_address and not private_key_str:
+            raise ProtostarException(
+                f"Private key has to be specified either with private-key-path option or {PRIVATE_KEY_ENV_VAR_NAME} "
+                "environment variable"
+            )
+
+        if (
+            not private_key_str or not args.account_address
+        ):  # FIXME(arcticae): This is temporary, when the signing is mandatory this should be removed
+            logger.warning(
+                "Signing declare transactions will be mandatory in future versions, please refer to the docs for "
+                "more details "
+            )
+            return None
+
+        try:
+            private_key = int(private_key_str, 16)
+        except ValueError as v_err:
+            raise ProtostarException(
+                f"Invalid private key format ({private_key_str}). Please provide hex-encoded number."
+            ) from v_err
+
+        key_pair = KeyPair.from_private_key(private_key)
+
+        try:
+            signer = PatchedStarkCurveSigner(
+                account_address=args.account_address,
+                key_pair=key_pair,
+                chain_id=network_config.chain_id,
+            )  # FIXME(arcticae): Change the default signer to starknet.py one, on 0.10 support
+        except ValueError as v_err:
+            raise ProtostarException(
+                f"Invalid account address format ({args.account_address}). Please provide hex-encoded number."
+            ) from v_err
+
+        return signer
+
+    @staticmethod
+    def _validate_signer_interface(signer_class):
+        if not issubclass(signer_class, BaseSigner):
```

Comment:
i am wondering if there's any reason we should be worried that somehow the imported class extends the wrong `BaseSigner`? I'm guessing that there's no possibility to import i.e. a different version of starknet.py in the same run, thus making interface different. @mkaput i would appreciate your insight on this

Reply:
I think that if somebody is importing stuff from external `starknet.py` module than ours, then they are highly probable to use something incompatible with our instance and thus we should reject such objects.

Reply:
I am wondering if such situation would even be possible in python's module system, for 2 modules of the same name to coexist

Reply:
> I am wondering if such situation would even be possible in python's module system, for 2 modules of the same name to coexist

I don't know, but current code it capable of dealing with both situations :)